lib/mvi/service.rb

```diff
@@ -31,15 +32,46 @@ def find_profile(user)
       raw_response = perform(:post, '', create_profile_message(user), soapaction: OPERATIONS[:find_profile])
       MVI::Responses::FindProfileResponse.with_parsed_response(raw_response)
     rescue Faraday::ConnectionFailed => e
-      Rails.logger.error "MVI find_profile connection failed: #{e.message}"
+      log_message_to_sentry("MVI find_profile connection failed: #{e.message}", :error)
       MVI::Responses::FindProfileResponse.with_server_error
     rescue Common::Client::Errors::ClientError, Common::Exceptions::GatewayTimeout => e
-      Rails.logger.error "MVI find_profile error: #{e.message}"
+      log_message_to_sentry("MVI find_profile error: #{e.message}", :error)
       MVI::Responses::FindProfileResponse.with_server_error
+    rescue MVI::Errors::Base => e
+      mvi_error_handler(user, e)
     end
 
     private
 
+    # TODO: Possibly consider adding Grafana Instrumentation here too
+    def mvi_error_handler(user, e)
+      case e
+      when MVI::Errors::DuplicateRecords
+        log_message_to_sentry('MVI Duplicate Record', :warn, error_context(user))
+      when MVI::Errors::RecordNotFound
```

Comment:
I can see duplicate records being a warn but is record not found? At least for MVI I think it's expected and part the flow through the app so maybe it should be at the info level?

Reply:
Good point on that one. Record not found could become really annoying.